src/bionics.cpp

```diff
@@ -1512,9 +1968,28 @@ bool player::install_bionics(const itype &type, int skill_level)
                                 difficult);
     }
 
-    if (!query_yn(
-            _("WARNING: %i percent chance of genetic damage, blood loss, or damage to existing bionics! Install anyway?"),
-            100 - chance_of_success)) {
+    //fixit: placeholder with hardcoded values is used temporary:
+    std::map<body_part, int> affected_bodyparts;
+    affected_bodyparts.emplace( bp_torso, 3 );
+    affected_bodyparts.emplace( bp_head, 1 );
+    affected_bodyparts.emplace( bp_mouth, 2 );
+
+    std::map<body_part, int> issues;
+    for( auto& elem : affected_bodyparts ) {
+        if( !has_enough_slots( elem.first, elem.second ) ) {
+            issues.insert( elem );
+        }
+    }
+
+    // moved out from loop above to show all requirements which are not satisfied
+    if( !issues.empty() ) {
+        //todo: show info from issues map in this popup:
+        popup( _( "Not enough space for bionic installation!" ) );
+        return false;
+    }
+
+    if( !query_yn( _( "WARNING: %i percent chance of genetic damage, blood loss, or damage to existing bionics! Continue anyway?" ),
+                   int( 100 - chance_of_success ) ) ) {
```

Comment:
When displaying negative effect chances, rounding up is preferable (`100 - int( chance_of_success )`).